From fed9937782eecc371821ae3c81bf0337005f3f0e Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Fri, 13 Jun 2014 14:35:44 -0400
Subject: [block] blk-mq: remove extra requeue trace

Message-id: <1402670238-13916-57-git-send-email-snitzer@redhat.com>
Patchwork-id: 83809
O-Subject: [RHEL7.1 PATCH 056/150] blk-mq: remove extra requeue trace
Bugzilla: 1105204
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Jeff Moyer <jmoyer@redhat.com>

BZ: 1105204

Upstream commit 74814b1c5569f5503727cf3052a52e2349818409
Author: Jens Axboe <axboe@fb.com>
Date:   Fri May 2 11:24:48 2014 -0600

    blk-mq: remove extra requeue trace

    We already issue a blktrace requeue event in
    __blk_mq_requeue_request(), don't do it from the original caller
    as well.

    Signed-off-by: Jens Axboe <axboe@fb.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/block/blk-mq.c b/block/blk-mq.c
index ca51ee4..0d37983 100644
--- a/block/blk-mq.c
+++ b/block/blk-mq.c
@@ -427,13 +427,9 @@ static void __blk_mq_requeue_request(struct request *rq)
 
 void blk_mq_requeue_request(struct request *rq)
 {
- struct request_queue *q = rq->q;
-
  __blk_mq_requeue_request(rq);
  blk_clear_rq_complete(rq);
 
- trace_block_rq_requeue(q, rq);
-
  BUG_ON(blk_queued_rq(rq));
  blk_mq_insert_request(rq, true, true, false);
 }
-- 
1.7.1